src/plugins/placeholder.js

```diff
@@ -15,7 +15,11 @@
     var brFiller = CKEDITOR.env.needsBrFiller ? '<br>' : '';
 
     var enterModeEmptyValue = {
-        1: ['<p>' + brFiller + '</p>'],
+        1: [
+            '<p>' + brFiller + '</p>',
+            '<h1>' + brFiller + '</h1>',
+            '<h2>' + brFiller + '</h2>'
```

Comment:
Hey @vbence86, I'd bet this is not enough... what about `<h3>`, or `<h4>` or ...

This bit of code has a lot of history. IIRC, this was the most robust we could come up with after a lot of trial and error. We used to just check if `editor.getData() === ''` or something like that.

Maybe this could be generalized?

Actually, from your steps, I can think of a different case that might fail: Apply a transformation that just adds attributes to the `p` or any other tag, so at the end, the empty content looks like `<p data-foo="bar"></p>`.

Reply:
Absolutely, I had all intention to make this way more generic than it is now, added a piece of code that looped through all the elements in the editor, excluded `<br>` elements along with all other whitespace characters and determined whether the editor's content is empty or not. Despite having worked pretty good, few of the tests cases began to fail due to unexpected appearance of the `data-placeholder` text. At that point I understood that few of the content tags that can be placed into the editor had a visual representation regardless the content of them. For example.: `<code>`, `<pre>`, `<img>` and so on. As you pointed out there are more than just a few scenarios when the current approach fails. 
```
<p></p>
<p><br><br></p>
<p><br><br><br></p>
<h1></h1>
<h1><br><br></h1>
<h2></h2>
<h2><br><br></h2>
```

Also should we regard multiple `<br>` (e.g.: `<p><br><br><br><br></p>`) as an intended way to add spacing between elements and therefore make the placeholder disappear? 

There are lot clarify if we want this to be more universal. I'll open an issue for refactoring this logic and create a pull request with my intial regex solution. In the meantime, in order to ensure velocity for 1.5.0 we could carry on with the extended version of the current solution.

And lastly as for other tags like `<h3>`, I haven't found any way to reproduce this issue with `<h3>`, `<h4>` etc. headers. These headers cannot be selected through the inline toolbox.

![captura de pantalla 2017-11-07 a las 10 59 31](https://user-images.githubusercontent.com/6104164/32487764-e373ed22-c3aa-11e7-91a1-4cbadcb4944c.png)